src/python/pants/engine/streaming_workunit_handler.py

```diff
@@ -185,30 +197,33 @@ def __init__(
         self.max_workunit_verbosity = max_workunit_verbosity
 
     def start(self) -> None:
```

Comment:
The `StreamingWorkunitHandler` code gets refactored quite a bit in a followup PR to have a `_CallbackThread` (currently known as `_InnerHandler`) per each callback. The rest of the PR is fairly stable.